src/main/com/fulcrologic/fulcro/dom.cljs

```diff
@@ -155,23 +155,50 @@
           (update-state
             this (.-props this)
             (gobj/getValueByKeys event "target" "value"))))
+      (componentDidUpdate [this prev-props prev-state snapshot]
+        (let [state-value    (gobj/getValueByKeys this "state" "value")
+              this-node      (js/ReactDOM.findDOMNode this)
+              value-node     (if (is-form-element? this-node)
+                               this-node
+                               (gdom/findNode this-node #(is-form-element? %)))
+              element-value  (gobj/get value-node "value")
+
+              props          (.-props this)
+
+              ;; diff the old and new props to determine if any changed or were added/removed
+              ;; value and onChange are elided because they always change when the field is updated
+              diff           (clojure.data/diff (js->clj prev-props) (js->clj props))
+              old            (-> (first diff)
+                               (dissoc "value")
+                               (dissoc "onChange"))
+              new            (-> (second diff)
+                               (dissoc "value")
+                               (dissoc "onChange"))
+              props-changed? (or (not-empty new) (not-empty old))
+
+              next-props     (if props-changed?
+                               (let [new-props #js {}]
+                                 (gobj/extend new-props props)
+                                 new-props)
+                               props)]
+
+          ;; delete keys for props that went away
+          ;; ... setState is always a merge and the only way to "delete" keys is to set to nil
+          (when props-changed?
+            (doseq [old-k (gobj/getKeys prev-props)]
+              (when-not (gobj/get next-props old-k)
+                (gobj/set next-props old-k nil))))
 
-      (UNSAFE_componentWillReceiveProps [this new-props]
```

Comment:
Nope...you have to use the UNSAFE variant or you'll get React warnings (and even crashes) in newer versions...this one is going away.